rclcpp_lifecycle/include/rclcpp_lifecycle/lifecycle_node_impl.hpp

```diff
@@ -119,16 +120,24 @@ LifecycleNode::create_client(
 }
 
 template<typename ServiceT, typename CallbackT>
-typename rclcpp::Service<ServiceT>::SharedPtr
+typename rclcpp_lifecycle::LifecycleService<ServiceT>::SharedPtr
 LifecycleNode::create_service(
   const std::string & service_name,
   CallbackT && callback,
   const rmw_qos_profile_t & qos_profile,
   rclcpp::CallbackGroup::SharedPtr group)
 {
-  return rclcpp::create_service<ServiceT, CallbackT>(
-    node_base_, node_services_,
-    service_name, std::forward<CallbackT>(callback), qos_profile, group);
+  rclcpp::AnyServiceCallback<ServiceT> any_service_callback;
+  any_service_callback.set(std::forward<CallbackT>(callback));
+
+  rcl_service_options_t service_options = rcl_service_get_default_options();
+  service_options.qos = qos_profile;
+
+  auto service = LifecycleService<ServiceT>::make_shared(
+    node_base_->get_shared_rcl_node_handle(), service_name, any_service_callback, service_options);
+  auto serv_base_ptr = std::dynamic_pointer_cast<rclcpp::ServiceBase>(service);
+  node_services_->add_service(serv_base_ptr, group);
+  return service;
```

Comment:
Is it possible to replace this code with something similar to https://github.com/ros2/rclcpp/blob/bba7c9fd144525dd4f2bba39fda6f72848a54db4/rclcpp_lifecycle/include/rclcpp_lifecycle/lifecycle_node_impl.hpp#L52-L57?

Reply:
I considered it, and I did not do it because I thought I had to change the create_service template of rclcpp, which is somewhat far from the purpose of this PR. 

Unlike `create_publisher`, the return type of `create_service` is a specific type `rclcpp::Service`.
https://github.com/ros2/rclcpp/blob/7a2ee2328151ebbc29ea3aed611b231eacbd51d8/rclcpp/include/rclcpp/create_publisher.hpp#L45-L46
https://github.com/ros2/rclcpp/blob/7a2ee2328151ebbc29ea3aed611b231eacbd51d8/rclcpp/include/rclcpp/create_service.hpp#L33-L34

In the next PR, it is planned to change the create_service part to a factory design like a publisher. https://github.com/ros2/rclcpp/pull/1836/files#r767374194
At that time, I will also consider changing the template parameter from rclcpp::Service<ServiceT> to ServiceT as well.